server/src/server.ts

```diff
@@ -163,6 +163,12 @@ disposables.push(
       return getEmbeddedLanguageTypeOnPosition(uriString, position)
     }
   ),
+  connection.onRequest(

```

Comment:
I'm a bit worried seeing many new requests for such minimal features. Requests are complex to understand and add a lot of overhead. They also increase the client/server mixed responsibility nightmare. I'd prefer a more minimal support, with more false negatives, if it could be implemented fully on the client side.

Reply:
I removed all the new requests, as I realized I was actually getting a lot of flickering when typing fast.